### PR TITLE
Enables api and sentinel e2e tests 

### PR DESCRIPTION
**Pull request by dianabarsan**
# Description

Takes a different approach to enabling/disabling protractor control flow and performs switch at the lowest level, and reduce the number of lines of changes to a minimum. 

Cherry picks a commit from `master` to fix an e2e test. 

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by dianabarsan** [comment]
hey @newtewt and @craig-landry 
I'm trying a new approach of enabling of disabling the control flow and how it affect internal methods.

After spending some hours into migrating the API e2e test to using the `native` suffixed functions (PR ended up having 1000 lines of changes just with those switches!), and then doing the same for the Sentinel tests (similar outcomes) and thinking we'd have to switch them back again when we remove the test with the enabled control flow (I presume we don't want all of our functions to have a "native" suffix forever). 

So, what I'm doing is pretty hacky. I'm setting an environment variable in both configuration files (conf.js and conf-disabled.js) and checking this variable in `utils.js` and updating the `request` function - which is then used by **everything else** - to use or not use the protractor promise or default promises. 
This means that we don't need to change any utils functions and not even have the `native` suffix. 

Please tell me what you think :) 
Thanks!

**Comment by dianabarsan** [comment]
Also, I don't think we should start switching the api and sentinel e2e tests to async await. It'd be a pointless effort and require thousands of line changes.

File: sentinel/src/lib/feed.js [hunk 1]
```diff
@@ -34,7 +34,11 @@ const getTransitionSeq = () => {
     });
 };
 
-const registerFeed = seq => {
+const registerFeed = (seq) => {
```
Comment (dianabarsan) [inline sentinel/src/lib/feed.js:37]: These were cherry picked from https://github.com/medic/cht-core/commit/3425c67b1fc11ee2ea879c5194925ef9dc2bf538

File: sentinel/tests/unit/lib/feed.js [hunk 2]
```diff
@@ -203,6 +204,26 @@ describe('feed', () => {
         });
     });
 
+    it('should restart the queue', () => {
```
Comment (dianabarsan) [inline sentinel/tests/unit/lib/feed.js:207]: These were cherry picked from https://github.com/medic/cht-core/commit/3425c67b1fc11ee2ea879c5194925ef9dc2bf538

**Comment by craig-landry** [comment]
> Also, I don't think we should start switching the api and sentinel e2e tests to async await. It'd be a pointless effort and require thousands of line changes. They work fine as is, with standard promise chains.

Yup, agreed. Things that properly manage their promises (don't rely on the weird promise manager) are totally fine.

**Comment by dianabarsan** [comment]
> Yup, agreed.

In that case, we can enable more immediately, without further changes.

**Review by craig-landry — APPROVED** [review]

**Comment by craig-landry** [comment]
> As an addendum: I'm not talking about tests that handle UI here, I'm talking about tests that already managed promises correctly and didn't rely on the protractor control flow. (like api, sentinel, transitions etc)
> 
> It means that we can remove:
> `setUserContactDocNative`, `revertDbNative`, `deleteUsersNative` etc, all this duplicated code from tests/utils and have the tests just use the already existent functions. And no need to update them when we fully remove control flow.

Given the "addendum" this sounds good to me. Our goal on this effort is getting everything off the promise manager. This technique works well for this group of tests and does it with less churn.

**Comment by dianabarsan** [comment]
> does it with less churn.
my particular pain was with duplicating line of code in that utils file without actually needing to.

**Comment by dianabarsan** [comment]
I'll follow up with a PR that removes all that duplicated code.

**Pull request merged**